### PR TITLE
Add adjustable semaphore

### PR DESCRIPTION
**Pull request by torstentrompler**

### CODE REVIEW

File: org.eclipse.scout.rt.platform.test/src/test/java/org/eclipse/scout/rt/platform/util/concurrent/AdjustableSemaphoreTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  public void testSetMaximumPermits_invalidInputs() {
+    final AdjustableSemaphore adjustableSemaphore = new AdjustableSemaphore(5);
+
+    assertThrows(IllegalArgumentException.class, () -> adjustableSemaphore.setMaximumPermits(0));
+    assertThrows(IllegalArgumentException.class, () -> adjustableSemaphore.setMaximumPermits(-1));
+    assertThrows(IllegalArgumentException.class, () -> adjustableSemaphore.setMaximumPermits(-10));
+  }
+
+  @Test
+  public void testSetMaximumPermits() {
```
Comment (andibur) [inline org.eclipse.scout.rt.platform.test/src/test/java/org/eclipse/scout/rt/platform/util/concurrent/AdjustableSemaphoreTest.java:47]: Could you also add some test cases that change the maximumNumber of permits while other parties have already acquired some or all of them before. Something like:

```
final AdjustableSemaphore adjustableSemaphore = new AdjustableSemaphore(5);
assertEquals(5, adjustableSemaphore.getMaximumPermits());
assertTrue(adjustableSemaphore.tryAcquire());
assertTrue(adjustableSemaphore.tryAcquire());
assertTrue(adjustableSemaphore.tryAcquire());
// now 3 of 5 permits are in use

adjustableSemaphore.setMaximumPermits(3);
assertEquals(3, adjustableSemaphore.getMaximumPermits());
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release();
assertTrue(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release(2);
// now 1 of 3 permits are in use

adjustableSemaphore.setMaximumPermits(2);
assertTrue(adjustableSemaphore.tryAcquire());
assertFalse(adjustableSemaphore.tryAcquire());
// now 2 of 2 permits are in use



// maybe another test case
final AdjustableSemaphore adjustableSemaphore = new AdjustableSemaphore(5);
assertTrue(adjustableSemaphore.tryAcquire(5));

adjustableSemaphore.setMaximumPermits(2);
// now 5 of 2 permits are in use
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release();
// 4 of 2 permits are in use
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release(2);
// 2 of 2 permits are in use
assertFalse(adjustableSemaphore.tryAcquire());

adjustableSemaphore.release();
// 1 of 2 permits are in use
assertTrue(adjustableSemaphore.tryAcquire());
```

**Review by andibur — APPROVED** [review]
Thanks

File: org.eclipse.scout.rt.platform.test/src/test/java/org/eclipse/scout/rt/platform/util/concurrent/AdjustableSemaphoreTest.java (hunk 1 above)
Comment (torstentrompler) [reply org.eclipse.scout.rt.platform.test/src/test/java/org/eclipse/scout/rt/platform/util/concurrent/AdjustableSemaphoreTest.java:47]: Good idea. Added two additional test cases for increasing/decreasing the permits.

**Review by andibur — APPROVED** [review]
nice, thanks

**Pull request merged**